### DISCUSSION TITLE
Leaflet limitation on terriajs?

### DISCUSSION THREAD

**Discussion by felixgismap** [TerriaJS/terriajs#6350] [Q&A]
Hi all,

I'm trying to understand which GIS platform to use for a project (most of the data in 2d: 70% 2d, 30% 3d).

Are there any limitations on the leaflet (when used in terriajs or not), compared to maplibre (Advanced vector editing, styling, actions when mouse hover on vector feature etc.)?

Thanks for the help.

**Comment by nf-s** [comment]
HI @felixgismap 

Leaflet doesn't support any 3D data, and doesn't support mapbox vector tiles without using an extension. The vector tile extensions are quite basic. Not sure which versions of Leaflet and how much of the mapbox style spec they support 

Maplibre has limited 3D data support, eg 2.5D vector data and terrain. I think there might be extensions to add 3d-tiles and other 3d formats. I haven't had any experience with mapbox-gl for 3-4 years (before version 2).

Mouse events are practically the same for both. Neither include vector editing - you would need to look into extensions. Both include vector styling - maplibre supports "native" mapbox vector tile style (pre version 2 anyway)

Leaflet in TerriaJS supports all our 2D data - including vector tiles. All 3D data must be viewed in Cesium
